### DISCUSSION TITLE
Building problems due to jsdelivr outages

### DISCUSSION THREAD

**Discussion by pchemguy** [sqlpage/SQLPage#478] [Q&A]
Since yesterday, I am having serious troubles with jsdelivr, and the SQLPage build script does not handle this (rare/unlikely?) issue in a robust fashion. (And I mean that this problem occurs with both my "amended" build process and the regular build process `cargo build` or `cargo build --profile superoptimized`.) The build script gets essentially locked without any feedback at the end while attempting to download the js/css assets. I had an instance where it froze for more than 12 min. I also independently (of the build script) confirmed jsdelivr problems by attempting to browse it and downloading the target assets manually. I am not sure whether this is just me, my location, or a global issue.

**Comment by lovasoa** [comment]
I had the same issues while I was in Russia. It seems to work fine in the rest of the world. I already reported the issue upstream in awc. In the meantime, you can manually download the assets and place them in the target directory: https://github.com/NixOS/nixpkgs/issues/310436#issuecomment-2194995358

**Reply by lovasoa** [reply]
@pchemguy , since I cannot reproduce the issue anymore, maybe you can help investigate it in https://github.com/actix/actix-web/issues/3382

**Reply by pchemguy** [reply]
@lovasoa, you are saying in that issue you could download the assets directly without problems. As I mentioned in my report, I observed the same connectivity problems when tried to browse the website and download the resources. I did not wait for the browsing failure, but it also kept trying to access resources for while without reporting a connection failure. And sometimes it reported failed a connection right away. So the problem might be country specific, but I do not think it is application/library specific. It very well might be related to the state activities aimed at filtering/eavesdropping the traffic.
